clippy_lints/src/no_effect_replace.rs

```diff
@@ -0,0 +1,89 @@
+use clippy_utils::diagnostics::span_lint;
+use clippy_utils::ty::is_type_diagnostic_item;
+use clippy_utils::SpanlessEq;
+use if_chain::if_chain;
+use rustc_ast::LitKind;
+use rustc_hir::{Expr, ExprKind};
+use rustc_lint::{LateContext, LateLintPass};
+use rustc_session::{declare_lint_pass, declare_tool_lint};
+use rustc_span::sym;
+
+declare_clippy_lint! {
+    /// ### What it does
+    /// Checks for `replace` statements which have no effect.
+    ///
+    /// ### Why is this bad?
+    /// It's either a mistake or confusing.
+    ///
+    /// ### Example
+    /// ```rust
+    /// "1234".replace("12", "12");
+    /// "1234".replacen("12", "12", 1);
+    /// ```
+    #[clippy::version = "1.62.0"]
+    pub NO_EFFECT_REPLACE,
+    suspicious,
+    "replace with no effect"
+}
+declare_lint_pass!(NoEffectReplace => [NO_EFFECT_REPLACE]);
+
+impl LateLintPass<'_> for NoEffectReplace {
+    fn check_expr(&mut self, cx: &LateContext<'_>, expr: &'_ Expr<'_>) {
+        let (path, args) = match expr.kind {
+            ExprKind::MethodCall(path, args, _) => (path, args),
+            _ => return,
+        };
+
+        if !(args.len() == 3 || args.len() == 4)
+            || !(path.ident.name.as_str() == "replace" || path.ident.name.as_str() == "replacen")
```

Comment:
This lint probably fits really good in the `methods` module. In that module you won't need all of that boilerplate.

Reply:
Change under that folder, it takes me many time, sorry for delay :)